examples_test.go

```diff
@@ -121,3 +122,50 @@ func ExampleConfig_whitelist() {
 	msg := <-consumer.Messages()
 	fmt.Fprintf(os.Stdout, "%s/%d/%d\t%s\t%s\n", msg.Topic, msg.Partition, msg.Offset, msg.Key, msg.Value)
 }
+
+// This example shows how to use the consumer to
+// consume msg from saram.Producer
+func ExampleConsumer_SaramProducer(){
+    config := cluster.NewConfig()
+
+    config.Group.Return.Notifications = true
+    brokers := []string{"127.0.0.1:9092"}
+    topic := "my_topic"
+
+    topics := []string{topic}
+    consumer, err := cluster.NewConsumer(brokers, "my-consumer-group", topics, config)
+    if err != nil {
+        panic(err)
+    }
+
+    // consumer really init successful when the notifications is received and `ntf.Type==cluster.RebalanceOK`
```

Comment:
Sorry for not noticing this earlier, but this is really misleading and prone to race conditions, as the consumer may rebalance (silently). I highly encourage you to switch to https://github.com/Shopify/sarama/pull/1099 which solves this problem.